src/components/InventoryTable/InventoryTable.test.js

```diff
@@ -1,296 +1,616 @@
-        <Router>
-          <InventoryTable hasAccess={false} isFullView />
-        </Router>
-      </Provider>
-    );
+const checkPagination = (perPage, page, total) => {
+  try {
+    const currentRange =
+      total === 0
+        ? '0 - 0'
+        : `${perPage * (page - 1) + 1} - ${Math.min(perPage * page, total)}`;
     expect(
-      toJson(wrapper.find('ForwardRef').first(), { mode: 'shallow' })
-    ).toMatchSnapshot();
-  });
-
-  it('should render correctly with items', () => {
-    const store = mockStore(initialState);
-    const wrapper = mount(
-      <Provider store={store}>
-        <Router>
-          <InventoryTable
-            items={[{ id: 'someId' }]}
-            page={5}
-            perPage={20}
-            total={200}
-            sortBy={{
-              index: 1,
-              key: 'one',
-              direction: 'asc',
-            }}
-          />
-        </Router>
-      </Provider>
-    );
+      screen.queryByTestId('inventory-table-top-toolbar')
+    ).toHaveTextContent(`${currentRange} of ${total}`);
     expect(
-      toJson(wrapper.find('ForwardRef').first(), { mode: 'shallow' })
-    ).toMatchSnapshot();
-  });
+      screen.queryByTestId('inventory-table-bottom-toolbar')
+    ).toHaveTextContent(`${currentRange} of ${total}`);
+  } catch (error) {
+    throw new Error(`Failed verifying pagination.\n\n${error}`);
+  }
+};
 
-  it('should render correctly with items no totla', () => {
-    const store = mockStore(initialState);
-    const wrapper = mount(
-      <Provider store={store}>
-        <Router>
-          <InventoryTable
-            items={[{ id: 'someId' }]}
-            page={5}
-            perPage={20}
-            sortBy={{
-              index: 1,
-              key: 'one',
-              direction: 'asc',
-            }}
-          />
-        </Router>
-      </Provider>
-    );
-    expect(
-      toJson(wrapper.find('ForwardRef').first(), { mode: 'shallow' })
-    ).toMatchSnapshot();
+const mockStore = configureStore([]);
+
+const initialState = {
+  entities: {
+    loaded: false,
+    tagsLoaded: false,
+    allTagsLoaded: false,
+    operatingSystems: [],
+    operatingSystemsLoaded: false,
+    groups: [],
+    invConfig: {},
+    sortBy: { key: 'updated', direction: 'desc' },
+  },
+};
+
+const emptyState = {
+  entities: {
+    ...initialState.entities,
+    activeFilters: [],
+    loaded: true,
+    rows: [],
+    page: 1,
+    perPage: 20,
+    count: 0,
+    total: 0,
+  },
+};
+
+const hostsPayload = buildHostsPayload();
+
+const loadedState = {
+  entities: {
+    ...initialState.entities,
+    activeFilters: [],
+    loaded: true,
+    ...{
+      ...hostsPayload,
+      rows: hostsPayload.results,
+    },
+  },
+};
+
+const errorState = {
+  entities: {
+    ...initialState.entities,
+    error: new Error('Loading error'),
+  },
+};
+
+describe('InventoryTable', () => {
+  useFeatureFlag.mockReturnValue(true); // assuming all features are enabled
```

Comment:
Defaulting to disabling all features might be better. When introducing a new feature all tests might be affected. But with them disabled by default, we can enable them only in tests related to that feature and still verify that other parts still work as intended.